### PR TITLE
vendor dskit 0ef13fb for httpgrpc tracing

### PR DESCRIPTION
**Pull request by francoposa**
#### What this PR does

Vendors in merged PR code from https://github.com/grafana/dskit/pull/353

applying the middleware in Mimir will be a separate PR

Do we update Mimir changelog whenever dskit is updated?

#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by pstibrany** [comment]
> LGTM.
> 
> > Do we update Mimir changelog whenever dskit is updated?
> 
> Yes please - please mention the tracing improvements as well as the changes from [grafana/dskit#364](https://github.com/grafana/dskit/pull/364) (I'd say something like "[BUGFIX] Querier: cancel query requests to ingesters as soon as any ingester in the same zone responds with an error").

We only include changes relevant to someone running (operating) Mimir in the CHANGELOG. If user should care about "Querier: cancel query requests to ingesters", I would suggest to rephrase it as optimization / enhancement. I'm not sure it's a bugfix.

**Comment by francoposa** [comment]
I added a changelog entry for the DoUntilQuorum enhancements; but I will save the tracing changeelog entry for the PR where we actually apply the middleware, since an operator won't see any change to tracing just from the vendoring in this PR

File: CHANGELOG.md [hunk 1]
```diff
@@ -70,6 +70,7 @@
 * [ENHANCEMENT] Ingester: reduce memory usage of active series tracker. #5665
 * [ENHANCEMENT] Store-gateway: added `-store-gateway.sharding-ring.auto-forget-enabled` configuration parameter to control whether store-gateway auto-forget feature should be enabled or disabled (enabled by default). #5702
 * [ENHANCEMENT] Querier: improved observability of calls to ingesters during queries. #5724
+* [ENHANCEMENT] Querier: cancel contexts for requests to ingesters in a zone upon first error received from the zone. #5755
```
Comment (charleskorn) [inline CHANGELOG.md:73]:
```suggestion
* [ENHANCEMENT] Querier: cancel query requests to ingesters in a zone upon first error received from the zone, to reduce wasted effort spent computing results that won't be used. #5755
```

**Comment by pracucci** [comment]
Sorry @francoposa, we just merged this PR https://github.com/grafana/mimir/pull/5764 which upgraded dskit. Could you rebase your PR, please?

**Comment by francoposa** [comment]
@pracucci no need now - that dskit PR included my changes, since it was the commit ahead of mine

**Pull request closed without merging**